Add SARIF parser

tushar-deepsource opened the pull request:
- Adds independent `sarif-parser` packaage, with tests and a CLI.
- Adds a CLI in `run_community_analyzer.py` which passes the correct `issue_map.json`, and a test for artifacts.
- Adds tox config and CI with reporting test coverage to DeepSource.

tushar-deepsource committed:
Add SARIF parser

enterprise-deepsource-icu[bot] committed:
style: format code with isort and Black

This commit fixes the style issues introduced in fe6def4 according to the output
from isort and Black.

Details: https://github.com/DeepSourceCorp/community-analyzers/pull/3

deepsource-autofix[bot] committed:
style: format code with Black and isort

This commit fixes the style issues introduced in fe6def4 according to the output
from Black and isort.

Details: https://github.com/DeepSourceCorp/community-analyzers/pull/3

tushar-deepsource committed:
bugfix

tushar-deepsource committed:
add checkout action

tushar-deepsource committed:
Add run_community_analyzer.py and tests

tushar-deepsource committed:
Move tox and mypy scripts out

tushar-deepsource committed:
Update README and fix tox

tushar-deepsource committed:
Update tests glob

Signed-off-by: Tushar Sadhwani <86737547+tushar-deepsource@users.noreply.github.com>

tushar-deepsource committed:
Fix DeepSource issues

srijan-deepsource commented on sarif-parser/tests/cli_test.py:
```diff
@@ -0,0 +1,143 @@
+        },
+    },
+    {
+        "issue_code": "container-resources",
```
Is this made up, or are we missing this in the mapping?

tushar-deepsource replied:
Yes, that is made up. `sarif-parser` tests don't deal with real world mappings. Tests regarding that will be in the top level `tests` folder.

srijan-deepsource commented on tests/community_analyzer_test.py:
```diff
@@ -0,0 +1,29 @@
+        result = json.load(file)
+
+    assert result == {
+        "issues": [],  # TODO
```
There's a TODO comment here.

Also, did our Python Analyzer stop catching these?

srijan-deepsource commented:
Also, set up test coverage reporting?

tushar-deepsource commented:
> Also, set up test coverage reporting?

Don't have the perms to set secrets on this repo.

> There's a TODO comment here.

Can you give an actual kube linter artifact? I'll write a test against that.

> did our Python Analyzer stop catching these?

I think it only catches it if it's `#TODO: ...`

tushar-deepsource committed:
Add kubelinter test

srijan-deepsource commented:
> Don't have the perms to set secrets on this repo.

Added `DEEPSOURCE_DSN` as a secret. Go ahead and make changes in the workflow to report it.

> I think it only catches it if it's #TODO: ...

Can you verify and create issue? It should catch blanket TODOs as well.

tushar-deepsource commented:
It does get raised on `# TODO` as well. Not sure why it didn't happen in this case.

```python
def foo():
    pass # TODO
```
```console
$ python -m marvin_py -sl /tmp/code/**/*.py            
{"path": "/tmp/code/a.py", "line": 1, "column": 0, "message-id": "PYL-C0104", "message": "Disallowed name \"foo\"", "context": null}
{"path": "/tmp/code/a.py", "line": 2, "column": 10, "message-id": "PYL-W0511", "message": "TODO", "context": null}
```

tushar-deepsource committed:
Update tox.yml

Signed-off-by: Tushar Sadhwani <86737547+tushar-deepsource@users.noreply.github.com>

tushar-deepsource committed:
Remove duplicate setup python step

Signed-off-by: Tushar Sadhwani <86737547+tushar-deepsource@users.noreply.github.com>

tushar-deepsource committed:
Update tox.ini

Signed-off-by: Tushar Sadhwani <86737547+tushar-deepsource@users.noreply.github.com>

tushar-deepsource committed:
Add more tests

tushar-deepsource committed:
Docstrings

deepsource-autofix[bot] committed:
style: format code with Black and isort

This commit fixes the style issues introduced in 1520982 according to the output
from Black and isort.

Details: https://github.com/DeepSourceCorp/community-analyzers/pull/3

tushar-deepsource committed:
Add CLI test and better error type

tushar-deepsource committed:
no-cache-dir

tushar-deepsource committed:
remove unused imports

tushar-deepsource committed:
add pragma to exclude_lines

tushar-deepsource committed:
newline at the end

tushar-deepsource committed:
fix rcfile

srijan-deepsource commented on .github/workflows/tox.yml:
```diff
@@ -0,0 +1,36 @@
+      - name: Run tests and type checking
+        run: tox
+
+      - name: Report test coverage to DeepSource
```
use test cov action

tushar-deepsource replied:
Done

tushar-deepsource committed:
Use action
